asserts/pool_test.go

```diff
@@ -691,6 +878,396 @@ func (s *poolSuite) TestUpdate(c *C) {
+	c.Assert(err, IsNil)
+	c.Assert(ok, Equals, true)
+
+	toResolve, toResolveSeq, err = pool.ToResolve()
+	c.Assert(err, IsNil)
+	c.Check(toResolve, HasLen, 0)
+	c.Check(toResolveSeq, HasLen, 0)
+	c.Check(pool.Err("for_one"), IsNil)
+
+	err = pool.CommitTo(s.db)
+	c.Check(err, IsNil)
+	c.Assert(pool.Err("for_one"), IsNil)
+
+	// sequence point 1, revision 5 is still the latest.
+	_, err = s.seq1_1111r5.Ref().Resolve(s.db.Find)
+	c.Assert(err, IsNil)
+
+	// and sequence point 3 revision 5 wasn't added to asserts database.
+	_, err = s.seq3_1111r5.Ref().Resolve(s.db.Find)
+	c.Assert(asserts.IsNotFound(err), Equals, true)
+}
+
+func (s *poolSuite) TestUpdateSeqFormingPinnedNewerSequenceNewerRevisionNoop(c *C) {
+	assertstest.AddMany(s.db, s.hub.StoreAccountKey(""), s.seq1_1111r5)
+
+	pool := asserts.NewPool(s.db, 64)
+
+	atseq := &asserts.AtSequence{
+		Type:        s.seq1_1111r5.Type(),
+		SequenceKey: []string{"1111"},
+		Sequence:    s.seq1_1111r5.Sequence(),
+		Revision:    s.seq1_1111r5.Revision(),
+		Pinned:      true,
+	}
+	err := pool.AddSequenceToUpdate(atseq, "for_one") // group num: 0
+	c.Assert(err, IsNil)
+
+	toResolve, toResolveSeq, err := pool.ToResolve()
+	c.Assert(err, IsNil)
+	c.Check(toResolve, HasLen, 0)
+	c.Check(toResolveSeq, DeepEquals, map[asserts.Grouping][]*asserts.AtSequence{
+		asserts.MakePoolGrouping(0): {
+			&asserts.AtSequence{
+				Type:        s.seq1_1111r5.Type(),
+				SequenceKey: []string{"1111"},
+				Sequence:    1,
+				Revision:    5,
+				Pinned:      true,
+			}},
+	})
+
+	c.Check(pool.Err("for_one"), IsNil)
+
+	// resolve
+	ok, err := pool.Add(s.seq2_1111r7, asserts.MakePoolGrouping(0))
+	c.Assert(err, IsNil)
+	c.Assert(ok, Equals, true)
+
+	err = pool.CommitTo(s.db)
+	c.Check(err, IsNil)
+	c.Assert(pool.Err("for_one"), IsNil)
+
+	// sequence point 1, revision 5 is still the latest.
+	_, err = s.seq1_1111r5.Ref().Resolve(s.db.Find)
+	c.Assert(err, IsNil)
+
+	// and sequence point 2 revision 7 wasn't added to asserts database.
+	_, err = s.seq2_1111r7.Ref().Resolve(s.db.Find)
+	c.Assert(asserts.IsNotFound(err), Equals, true)
+}
+
+func (s *poolSuite) TestUpdateSeqFormingPinnedSameSequenceNewerRevision(c *C) {
+	assertstest.AddMany(s.db, s.hub.StoreAccountKey(""), s.seq1_1111r5)
+
+	pool := asserts.NewPool(s.db, 64)
+
+	atseq := &asserts.AtSequence{
+		Type:        s.seq1_1111r5.Type(),
+		SequenceKey: []string{"1111"},
+		Sequence:    s.seq1_1111r5.Sequence(),
+		Revision:    s.seq1_1111r5.Revision(),
+		Pinned:      true,
+	}
+	err := pool.AddSequenceToUpdate(atseq, "for_one") // group num: 0
+	c.Assert(err, IsNil)
+
+	toResolve, toResolveSeq, err := pool.ToResolve()
+	c.Assert(err, IsNil)
+	c.Check(toResolve, HasLen, 0)
+	c.Check(toResolveSeq, DeepEquals, map[asserts.Grouping][]*asserts.AtSequence{
+		asserts.MakePoolGrouping(0): {
+			&asserts.AtSequence{
+				Type:        s.seq1_1111r5.Type(),
+				SequenceKey: []string{"1111"},
+				Sequence:    1,
```

Comment:
minor nitpick, it's a little confusing to go back and forth between

```go 
Sequence: s.seq1_1111r5.Sequence()
```

and 
```go
Sequence: 1
```
even though they are the same value, it would be nice to be consistent with either providing the constant or the value from the seq variable for these

Reply:
Sure, fair point, I'll go over these variables and make them consistent.

Reply:
Done (hopefully I didn't miss any of those).